### PR TITLE
Fix #169: Skip tagging if the previous component does not exist

### LINKED ISSUE #169: Skip tagging if the previous component does not exist

### PR DESCRIPTION
**Pull request by jumde**
Fix #169

### CODE REVIEW

**Comment by jumde** [comment]
Verified that:

1. brave-core-ext-ad-block-update-publish-dev/79/ - PASSED
2. brave-core-ext-https-everywhere-update-publish-dev/14/ - PASSED
3. brave-core-ext-ipfs-component-publish-dev/52/ - UNRELATED FAILURE (INVESTIGATING)
4. brave-core-ext-local-data-files-update-publish-dev/21/ - PASSED
5. brave-core-ext-tor-client-update-publish-dev/11/ - PASSED
6. brave-core-ext-ethereum-remote-client-update-publish-dev/11/ - PASSED
7. brave-core-ext-ntp-sponsored-publish-dev/121/ - PASSED
8. brave-core-ext-ntp-super-referrer-publish-dev/22/ - PASSED
9. brave-core-ext-speedreader-data-files-update-publish-dev/15/ - PASSED

**Review by emerick — APPROVED** [review]
LGTM

File: lib/util.js [hunk 1]
```diff
@@ -211,25 +211,37 @@ const uploadExtension = (name, id, version, crxFile) => {
...
+          Key: `release/${id}/${previousComponentFilename}`
         }
-        aws_s3.putObjectTagging(params, function(err, data) {
-          if (err) {
-            console.error(`Tagging failed for ${id}/${previousComponentFilename} with tag ${getPreviousVersion(version).replace(/\./g, '.')}`)
-            console.error(err, err.stack)
-            reject(new Error('Failed to upload extension to S3'))
+
+        aws_s3.headObject(params, function (err, metadata) {
+          if (err && err.code === 'NotFound') {
```
Comment (yrliou) [inline lib/util.js:218]: With this new check, do we still need `if (version == FirstVersion) {` check above?

Comment (jumde) [reply lib/util.js:211]: Great catch! thanks @yrliou

**Review by yrliou — APPROVED** [review]
LGTM, thanks.

**Pull request merged**